paasta_tools/kubernetes/application/controller_wrappers.py

```diff
@@ -196,12 +200,71 @@ def create(self, kube_client: KubeClient) -> None:
         self.ensure_pod_disruption_budget(kube_client)
         self.sync_horizontal_pod_autoscaler(kube_client)
 
+    def deep_delete_and_create(self, kube_client: KubeClient) -> None:
+        self.logging.debug(
+            f"Deep Delete {self.kube_deployment.service} {self.kube_deployment.instance}."
+        )
+        self.deep_delete(kube_client)
+        self.logging.debug(
+            f"waiting for instance {self.kube_deployment.service} {self.kube_deployment.instance} to die."
+        )
+
+        timer = 0
+        while (
+            self.kube_deployment in set(list_all_deployments(kube_client))
+            and timer < 60
+        ):
+            sleep(1)
+            timer += 1
+
+        if timer >= 60 and self.kube_deployment in set(
+            list_all_deployments(kube_client)
+        ):
+            try:
+                force_delete_pods(
+                    self.item.metadata.name,
+                    self.kube_deployment.service,
+                    self.kube_deployment.instance,
+                    self.item.metadata.namespace,
+                    kube_client,
+                )
+            except ApiException as e:
+                if e.status == 404:
+                    # Deployment does not exist, nothing to delete but
+                    # we can consider this a success.
+                    self.logging.debug(
+                        "not deleting nonexistent deploy/{} from namespace/{}".format(
+                            self.kube_deployment.service, self.item.metadata.namespace
+                        )
+                    )
+                else:
+                    raise
+        else:
+            self.logging.info(
+                "deleted deploy/{} from namespace/{}".format(
+                    self.kube_deployment.service, self.item.metadata.namespace
+                )
+            )
+        self.logging.debug(
+            f"{self.kube_deployment.service} {self.kube_deployment.instance} are deleted, creating new one."
```

Comment:
Make sure to clean up logging. We already have longs in deep_delete and create. So you don't need so many logs. i think just logs in force_delte_pods and exception are enough. 
Looks good overall. Just remove redundant things, and test it again.

Reply:
ok only keeping the logging in the try/excpet.